src/lib/MSPVTX/devMSPVTX.cpp

```diff
@@ -0,0 +1,348 @@
+            if (power >= NUM_POWER_LEVELS)
+            {
+                power = 2; // 25 mW
+            }
+
+            channel = ((vtxConfigPacket->band - 1) * 8) + (vtxConfigPacket->channel - 1);   
+            if (channel >= FREQ_TABLE_SIZE)
+            {
+                channel = 27; // F4 5800MHz
+            }
+
+            if (vtxConfigPacket->bands == getFreqTableBands() && vtxConfigPacket->channels == getFreqTableChannels() && vtxConfigPacket->powerLevels == NUM_POWER_LEVELS)
+            {
+               mspState = CHECK_POWER_LEVELS;
+               break;
+            }
+            clearVtxTable();
+            break;
+        case MONITORING:
+            pitMode = vtxConfigPacket->pitmode;
+
+            // Set power before freq changes to prevent PLL settling issues and spamming other frequencies.
+            power = vtxConfigPacket->power - 1; // Correct for BF starting at 1.
+            vtxSPIPitmode = pitMode;
+            vtxSPIPowerIdx = power;
+
+            channel = ((vtxConfigPacket->band - 1) * 8) + (vtxConfigPacket->channel - 1);
+            if (channel < FREQ_TABLE_SIZE)
+            {
+                vtxSPIBandChannelIdx = channel;
+            }
+            break;
+        }
+        break;
+
+    case MSP_VTXTABLE_POWERLEVEL:
+        powerLevelPacket = (mspVtxPowerLevelPacket_t *)(packet + MSP_VTX_PAYLOAD_OFFSET);
+
+        switch (mspState)
+        {
+        case CHECK_POWER_LEVELS:
+            if (powerLevelPacket->powerValue ==  powerLevelsLut[checkingIndex] && powerLevelPacket->powerLabelLength == POWER_LEVEL_LABEL_LENGTH) // Check lengths before trying to check content
+            {
+                if (powerLevelPacket->label1 == powerLevelsLabel[checkingIndex * POWER_LEVEL_LABEL_LENGTH + 0] &&
+                    powerLevelPacket->label2 == powerLevelsLabel[checkingIndex * POWER_LEVEL_LABEL_LENGTH + 1] &&
+                    powerLevelPacket->label3 == powerLevelsLabel[checkingIndex * POWER_LEVEL_LABEL_LENGTH + 2])
+                {
+                    checkingIndex++;
+                    if (checkingIndex > NUM_POWER_LEVELS - 1)
+                    {
+                        checkingIndex = 0;
+                        mspState = CHECK_BANDS;
+                    }
+                    break;
+                }
+            }
+            setVtxTablePowerLevel(checkingIndex + 1);
+            break;
+        }
+        break;
+
+    case MSP_VTXTABLE_BAND:
+        bandPacket = (mspVtxBandPacket_t *)(packet + MSP_VTX_PAYLOAD_OFFSET);
+
+        switch (mspState)
+        {
+        case CHECK_BANDS:
+            if (bandPacket->bandNameLength ==  BAND_NAME_LENGTH && bandPacket->bandLetter == getBandLetterByIdx(checkingIndex) && bandPacket->isFactoryBand == IS_FACTORY_BAND && bandPacket->channels == CHANNEL_COUNT) // Check lengths before trying to check content
+            {
+                if (memcmp(bandPacket->bandName, channelFreqLabel + checkingIndex * CHANNEL_COUNT, 8) == 0)
```

Comment:
rather that specify 8 you should use sizeof(bandPacket->bandName).

Reply:
fixed